www/js/app.js

```diff
@@ -254,11 +254,33 @@ define(['jquery', 'abstractBackend', 'util', 'cookies','geometry','osabstraction
         $('#searchingForTitles').hide();
         return false;
     });
-    $('input:radio[name=contentInjectionMode]').on('change', function(e) {
+    $('input:radio[name=contentInjectionMode]').on('change', function(event) {
         // Do the necessary to enable or disable the Service Worker
         setContentInjectionMode(this.value);
         checkSelectedArchiveCompatibilityWithInjectionMode();
     });
+    $('#articleContent').on('load',onIFrameLoad);
+    function onIFrameLoad() {
+        $('#centeredspinner').hide();
+        $('#articleContent').contents().on('click', function (event) {
+            if (contentInjectionMode === 'serviceworker') {
+                // In Service Worker mode, we want to display a spinner
+                // when the user clicks on a link
+                // We only do that when the click is somewhere inside a <a> tag
```

Comment:
Is that really a safe way? We could wait for a message from the service worker (that is sent as soon as the service worker starts loading the article).